clients/admin-ui/src/features/taxonomy/components/TaxonomyInteractiveTree.tsx

```diff
@@ -0,0 +1,222 @@
+import "@xyflow/react/dist/style.css";
+
+import {
+  Background,
+  BackgroundVariant,
+  Controls,
+  Edge,
+  EdgeTypes,
+  MiniMap,
+  Node,
+  NodeTypes,
+  ReactFlow,
+  ReactFlowProvider,
+  useReactFlow,
+} from "@xyflow/react";
+import palette from "fidesui/src/palette/palette.module.scss";
+import { useEffect, useMemo } from "react";
+
+import { TAXONOMY_ROOT_NODE_ID } from "../constants";
+import { TaxonomyTreeHoverProvider } from "../context/TaxonomyTreeHoverContext";
+import useD3HierarchyLayout from "../hooks/useD3HierarchyLayout";
+import { TaxonomyEntity } from "../types";
+import { CoreTaxonomiesEnum } from "../types/CoreTaxonomiesEnum";
+import TaxonomyTextInputNode, {
+  TextInputNodeType,
+} from "./TaxonomyTextInputNode";
+import TaxonomyTreeEdge from "./TaxonomyTreeEdge";
+import TaxonomyTreeNode, { TaxonomyTreeNodeType } from "./TaxonomyTreeNode";
+
+interface TaxonomyInteractiveTreeProps {
+  taxonomyType: CoreTaxonomiesEnum;
+  taxonomyItems: TaxonomyEntity[];
+  draftNewItem: Partial<TaxonomyEntity> | null;
+  lastCreatedItemKey: string | null;
+  resetLastCreatedItemKey: () => void;
+  onTaxonomyItemClick: (taxonomyItem: TaxonomyEntity) => void;
+  onAddButtonClick: (taxonomyItem: TaxonomyEntity | undefined) => void;
+  onCancelDraftItem: () => void;
+  onSubmitDraftItem: (label: string) => void;
+}
+
+const TaxonomyInteractiveTree = ({
+  taxonomyType,
+  taxonomyItems,
+  draftNewItem,
+  lastCreatedItemKey,
+  resetLastCreatedItemKey,
+  onTaxonomyItemClick,
+  onAddButtonClick,
+  onCancelDraftItem,
+  onSubmitDraftItem,
+}: TaxonomyInteractiveTreeProps) => {
+  const { fitView } = useReactFlow();
+  useEffect(() => {
+    setTimeout(() => fitView(), 0);
```

Comment:
I've added some comments:
`
  // Reset the zoom level and center the view when the taxonomy type changes
  useEffect(() => {
    // Timeout is needed because fitView doesn't work if it's
    // called before the nodes are rendered
    setTimeout(() => fitView(), 0);
  }, [fitView, taxonomyType]);
`